src/student/student.service.ts

```diff
@@ -3,21 +3,96 @@ import { CreateStudentDto } from './dto/create-student.dto';
 import { UpdateStudentDto } from './dto/update-student.dto';
 import { ReadStudentDto } from './dto/read-student.dto';
 import { PrismaService } from 'src/prisma/prisma.service';
+import { createClient } from '@supabase/supabase-js';
+import { decode } from 'base64-arraybuffer';
+import { v4 as uuidv4 } from 'uuid';
 
 @Injectable()
 export class StudentService {
   constructor(private prisma: PrismaService) {}
 
-  create(createStudentDto: CreateStudentDto) {
-    const testStudent = this.prisma.testModel.create({
+  getPaymentById(createStudentDto) {
+    return this.prisma.payment.findUnique({
+      where: {
+        id: createStudentDto.paymentId,
+      },
+    });
+  }
+
+  async createPayment(payment_path: string, isRegisterByStudent: boolean) {
+    const newPayment = await this.prisma.payment.create({
+      data: {
+        photo_src: payment_path,
+        // if submission is by student, put in pending status
+        // until cashier accepts payment
+        // else submission is by cashier, accepted status
+        status: isRegisterByStudent ? 'pending' : 'accepted',
+      },
+    });
+    return newPayment;
+  }
+
+  async createEvent() {
+    const newEvent = await this.prisma.event.create({
+      data: {
+        title: '',
+        price: '',
+        max_participants: 1,
+        requires_payment: true,
+        description: '',
+        date: new Date(),
+        is_active: true,
+        form_name: '',
+      },
+    });
+    return newEvent;
+  }
+
+  async createStudent(
+    createStudentDto: CreateStudentDto,
+    file: Express.Multer.File,
+  ) {
+    const uuid = uuidv4();
+    const payment_path = this.uploadImageToDB(file, uuid);
+    createStudentDto.uuid = uuid;
```

Comment:
`createStudentDto.uuid = uuidv4()` nalang siguro dretso para wala na ng `const uuid = uuidv4()`

Reply:
Gi buhatan nako ang uuid ug own variable niya kay para ma pass siya sa `uploadImageToDB()` kay basig pag mag `this.uploadImageToDB(file, uuidv4());` ko kay mag create siyag lahi na uuid na dili parehas sa `createStudentDto.uuid` 

Tama ba?

Reply:
Ay wa nako na notice na ma pass siya sa uploadImageToDb. Tama na, mbmb thank u!

Reply:
Resolved, thanks!
Action:
-None